internal/validate/namespace_object.go

```diff
@@ -35,3 +40,102 @@ func getNamespaceObject(name string) ref.Val {
 
 	return types.NewDynamicMap(types.DefaultTypeAdapter, namespaceObjectData)
 }
+
+func getKubernetesResource(name string, namespace string, apiVersion string, kind string) (any, error) {
+	resourceRequest := kubernetes.GetResourceRequest{
+		APIVersion: apiVersion,
+		Kind:       kind,
+		Name:       name,
+		Namespace:  &namespace,
+	}
+	responseBytes, err := kubernetes.GetResource(&host, resourceRequest)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get Kubernetes resource: %w", err)
+	}
+
+	var response map[string]any
+	if err = json.Unmarshal(responseBytes, &response); err != nil {
+		return nil, fmt.Errorf("cannot unmarshal Kubernetes resource response: %w", err)
+	}
+
+	return response, nil
+}
+
+// This function is used by the policy to get the params using selector.
+func getKubernetesResourceList(
+	namespace string,
+	apiVersion string,
+	kind string,
+	selector *metav1.LabelSelector,
+) ([]any, error) {
+	labelSelectorString, err := formatLabelSelectorString(selector)
+	if err != nil {
+		return nil, err
+	}
+	resourceRequest := kubernetes.ListResourcesByNamespaceRequest{
+		APIVersion:    apiVersion,
+		Kind:          kind,
+		Namespace:     namespace,
+		LabelSelector: &labelSelectorString,
+	}
+
+	responseBytes, err := kubernetes.ListResourcesByNamespace(&host, resourceRequest)
+	if err != nil {
+		return nil, fmt.Errorf("cannot list Kubernetes resources: %w", err)
+	}
+
+	var response map[string]any
+	if err = json.Unmarshal(responseBytes, &response); err != nil {
+		return nil, fmt.Errorf("cannot unmarshal Kubernetes list resources response: %w", err)
+	}
+	items, ok := response["items"].([]any)
+	if !ok {
+		return nil, fmt.Errorf("cannot unmarshal Kubernetes list resources response: %w", err)
+	}
+	return items, nil
+}
+
+// Function used to format a metav1.LabelSelector into a string that can be understood
+// by Kubewarden host capabilities.
+func formatLabelSelectorString(selector *metav1.LabelSelector) (string, error) {
```

Comment:
are these functions reused from somewhere? (if so it would be nice to add a link and possibly reuse unit tests too).

Reply:
No, these are not reused functions.

Reply:
Thanks for the unit test! resolving.